### ISSUE TITLE
Image size constraints don't work properly

### ISSUE THREAD

**Issue by Qythyx** [dotnet/maui#14411]
### Description

There are a number of related problem here. I will try to walk through them all.

My goal is to show an image next to text such that the image is aspect scaled to be the same height as the text. In MAUI the image is, by default, shown in it's full size. In my case these icons are 512x512 pixels, which is way too big.

My first attempt was to something like the following in a Grid.
```xaml
	<Label
		x:Name="RatingLabel"
		Grid.Column="0"
		Margin="{DynamicResource MarginR}"
		LineBreakMode="NoWrap"
		Style="{DynamicResource LabelBody}"
		Text="{Binding Text, Source={x:Reference This}}"
		TextColor="#FFC000" />
	<Image
		x:Name="Stars"
		Grid.Column="1"
		Margin="{DynamicResource MarginR}"
		Aspect="AspectFit"
		MaximumHeightRequest="{Binding Height, Source={x:Reference RatingLabel}}"
		Source="{Binding RatingStarFile, Source={x:Reference This}}" />
```

I was hoping that the `Aspect="AspectFit"` and `MaximumHeightRequest` on the Image would constrain it to that height, and keep it at the correct aspect ratio. In iOS and Android this ended up not showing the image at all. I tried changing `MaximumHeightRequest` to `HeightRequest` but that didn't help. This solution feels ideal to me because it doesn't require anything extra in the code behind.

Next I  tried this:
```xaml
	<Image
		x:Name="Stars"
		Grid.Column="1"
		Margin="{DynamicResource MarginR}"
		HeightRequest="{Binding StarHeight, Source={x:Reference This}}"
		WidthRequest="{Binding StarWidth, Source={x:Reference This}}"
		Source="{Binding RatingStarFile, Source={x:Reference This}}" />
```

with this in the code behind:

```cs
	public StarRating()
	{
		InitializeComponent();
		RatingLabel.SizeChanged += RatingLabel_SizeChanged;
	}

	private void RatingLabel_SizeChanged(object? sender, EventArgs e)
	{
		OnPropertyChanged(nameof(StarHeight));
		OnPropertyChanged(nameof(StarWidth));
	}

	public double StarHeight => RatingLabel?.Height ?? 20;
	public double StarWidth => (RatingLabel?.Height ?? StarHeight) * 5;
```

Note, the reason for the default size of 20 is because the first time the binding calls `StarHeight` the referenced `RatingLabel` is null, so I need to return something. This solution works perfectly in iOS. In Android, however, it renders the stars with the initial size of 20 and then does not update to the correct height after the `RatingLabel` is sized. I added debug statements and `RatingLabel_SizeChanged` is being called and the binding call to `StarHeight` is being called again after the `OnPropertyChanged` call. I don't know why the size isn't being updated and I haven't found a full solution for Android.

Also note, in this case I know my images have an aspect ratio of exactly 5 : 1. So in my `StarWidth` property I just multiply the height by 5. This works for this case, but if I did not know the aspect ratio of the image, or if I dynamically changed images that have varying aspect ratios then this strategy wouldn't work and I can't think of a great solution.

Given my desire to scale the image to match the font size, another option would be to get the pixel size for scaled fonts. That is, when a font autoscales (due to the system's font scaling setting) you no longer know the pixel size for it. In my case the label's style sets the font size to 14. But, AFAICT, there is no way to get the pixel size for that.

### Steps to Reproduce

See description above.

### Link to public reproduction project repository

n/a

### Version with bug

6.0.312

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

all

### Did you find any workaround?

A partial workaround is shown above.

**Comment by hartez** [comment]
> Version with bug
> 6.0.312

Have you tried this on .NET 7 yet?

**Comment by Qythyx** [comment]
> Have you tried this on .NET 7 yet?

Yes, I'm on .NET 7 and have this problem. Sorry, I'm not at my computer at the moment so I don't know the exact version.

**Comment by Qythyx** [comment]
According to [this comment](https://learn.microsoft.com/en-us/answers/questions/1198744/how-to-size-an-image-to-match-the-size-of-a-label?comment=answer-1196687&page=1#comment-1252376) the root problem in Android is:

> because in Android's drawing mechanism, drawing and measuring are done separately. In Android, the measure method is executed again to measure the size of the control after the control is rendered, while in iOS, both are done simultaneously. Therefore, Android needs Task.Delay to wait for the measure method to execute, while iOS does not.

If this is accurate, it would be great if there was an override or event we could connect to act after the second measure occurs. That would allow me to remove the 100ms delay in the repro mentioned above and respond directly to the measure. Interestingly, there already are `Measure`, `MeasureOverride`, and `OnMeasure` methods that can be overridden, but in my testing none of these did what I need.

**Comment by Tframe** [comment]
I'm facing similar issue where images load fine on iOS, but aren't on Android when binding to height and width properties in the VM. As @Qythyx mentioned above, setting the HeightRequest and WidthRequest after waiting 100 ms works fine as a workaround.

**Comment by borrmann** [comment]
I have the same problem... On iOS this ->
```
<Grid ColumnDefinitions="1*,1*,1*">

            <Image Source="fox.png"
                   Grid.Column="1"
                   Aspect="AspectFit" />
        </Grid>
```
basically ignores the bounds of its column BUT this only happens when the original png is larger then the bounds. Maybe this didnt happen during your tests because the original images arent large enough and needed to be downscaled!
<img width="299" alt="Screenshot 2023-06-30 112535" src="https://github.com/dotnet/maui/assets/20472867/e6c2bc1a-30bb-480e-beb6-029ef3fd4363">
one. 
Putting the image inside another Layout e.g. a grid and it works.
```
<Grid ColumnDefinitions="1*,1*,1*">
     <Grid Grid.Column="1*">
            <Image Source="fox.png"
                   Aspect="AspectFit" />
     </Grid>
</Grid>
```
<img width="294" alt="Screenshot 2023-06-30 113008" src="https://github.com/dotnet/maui/assets/20472867/e9de2e3a-5133-4d95-b57e-5fef67590887">

**Comment by OliverDevsCodex** [comment]
> > Version with bug
> > 6.0.312
> 
> Have you tried this on .NET 7 yet?

problem still exists in .net 8 and it is a frustrating one for some type of projects

**Comment by mdjavedakhtar** [comment]
i just wasted 2 days due to this bug.
debug was not available as i was building for an androidTV and it was in remote location !!
